### PR TITLE
add additional arrow key and numberpad support

### PR DESCRIPTION
**Pull request by Fuseteam**
give the left arrow key the same functionality as the page up key
give the right arrow key the same functionality as the page down key
give numpad enter key the same functionality as the normal enter key
allow the input of number via the numpad

### CODE REVIEW

**Comment by Fuseteam** [comment]
@salty-horse i'm looking forward to your feedback

File: ibus.py [hunk 1]
```diff
@@ -105,16 +110,22 @@ def do_process_key_event(self, keyval, keycode, state):
                 self.preedit_string = self.preedit_string[:-1]
                 self.invalidate()
                 return True
-            elif keyval in num_keys:
-                index = num_keys.index(keyval)
+           elif keyval in num_keys:
```
Comment (salty-horse) [inline ibus.py]: There's a missing space in the indentation so this doesn't run.

```
  File "/usr/share/ibus-uniemoji/ibus.py", line 113
    elif keyval in num_keys:
```

You can catch any errors by running `ibus-daemon --xim -r -v`.

File: ibus.py [hunk 2]
```diff
@@ -105,16 +110,22 @@ def do_process_key_event(self, keyval, keycode, state):
...
+                index = num_keys.index(keyval)
```
Comment (salty-horse) [inline ibus.py]: Please remove trailing space at the end of the line

File: ibus.py [hunk 3]
```diff
@@ -105,16 +110,22 @@ def do_process_key_event(self, keyval, keycode, state):
...
-            elif keyval in num_keys:
-                index = num_keys.index(keyval)
+           elif keyval in num_keys:
+                index = num_keys.index(keyval) 
+                if self.set_lookup_table_cursor_pos_in_current_page(index):
+                    self.commit_candidate()
+                    return True
+                return False
+            elif keyval in numpad_keys:
+                index = numpad_keys.index(keyval)
```
Comment (salty-horse) [inline ibus.py]: Please remove trailing space at the end of the line

File: ibus.py [hunk 4]
```diff
@@ -46,6 +46,11 @@ def debug(*a, **kw):
     num_keys.append(getattr(IBus, str(n)))
 num_keys.append(getattr(IBus, '0'))
 del n
+numpad_keys = []
```
Comment (salty-horse) [inline ibus.py]: please add an empty line above this

File: ibus.py [hunk 5]
```diff
@@ -46,6 +46,11 @@ def debug(*a, **kw):
...
+for n in range(1, 10):
+    numpad_keys.append(getattr(IBus, 'KP_'+str(n)))
```
Comment (salty-horse) [inline ibus.py]: please add spaces around the `+`

**Comment by Fuseteam** [comment]
yes i find left and right usefull for scrolling to the pages, i use them myself, i'll see if i can add numpad arrows aswell

File: ibus.py (hunk 1 above)
Comment (Fuseteam) [reply ibus.py]: a missing space in the indentation? i'm assuming on line 113?

Comment (salty-horse) [reply ibus.py]: Yes. Where it says `elif keyval in num_keys:` the indentation is 11 characters instead of 12.

**Comment by Fuseteam** [comment]
@salty-horse can you check now? it appears numpad arrow up and down has some weird behavior for me when i try it, i'll probably should do that in another patch

**Comment by Fuseteam** [comment]
hmm my ibus is acting up maybe i'll file a commit for it after all, i may or may not have misused for ```ibus-daemon --xim -r -v``` which may or may not be the reason for the weird behavior on my machine

**Comment by Fuseteam** [comment]
yep it was just my ibus that was messed up :P

**Comment by Fuseteam** [comment]
@salty-horse feedback please

File: ibus.py [hunk 6]
```diff
@@ -123,16 +123,16 @@ def do_process_key_event(self, keyval, keycode, state):
                     self.commit_candidate()
                     return True
                 return False
-            elif keyval == IBus.Page_Up or keyval == IBus.KP_Page_Up or keyval == IBus.Left:
+             elif keyval == IBus.Page_Up or keyval == IBus.KP_Page_Up or keyval == IBus.Left or keyval == IBus.KP_Left:
```
Comment (salty-horse) [inline ibus.py]: Now you added a space before `elif`. Are you sure this even runs? Please run it and see if the uniemoji appears.

Comment (Fuseteam) [reply ibus.py]: huh? i did i misunderstand? and yeah i'm pretty sure it ran the last time i test 🤔

Comment (Fuseteam) [reply ibus.py]: weird i would've sworn it ran before i pushed 👀

Comment (Fuseteam) [reply ibus.py]: anyway fixed, sorry for the trouble @salty-horse

**Comment by salty-horse** [comment]
At this point, all of the `keyval` checks are really lengthy. Can you change them to `keyval in (A, B, C)`?

**Comment by Fuseteam** [comment]
@salty-horse i changed them for all arrow key val checks

**Comment by Fuseteam** [comment]
oops uhh my mistake

**Comment by Fuseteam** [comment]
@salty-horse anything else i need to do?

**Comment by salty-horse** [comment]
Looks good! I pushed it to master as one commit fb4fd5fbc3cee5e673e4e0b87e695bb26b9d4a0d.

Thanks for the change, and the patience :)

**Comment by Fuseteam** [comment]
no, thank **you** for the patience and the tips :)

**Pull request closed without merging**